core/chain-configs/src/client_config.rs

```diff
@@ -301,6 +301,11 @@ pub fn default_produce_chunk_add_transactions_time_limit() -> Option<Duration> {
     Some(Duration::from_millis(200))
 }
 
+pub fn default_orphan_state_witness_pool_size() -> usize {
+    // With 4 shards, a capacity of 12 witnesses allows to store 3 orphan witnesses per shard.
+    12
+}
```

Comment:
Is it ok to spend 1GB of memory on the orphan cache?
It's not that much by itself, but there are many caches, it could add up to a large value in the end.

Reply:
It's fiiine... We will get rid of large trie cache worth of 1 or 3 GB so we can use this space.
But anyway, IMO pool size should be equal to `num_shards * range_len`, otherwise some witnesses may be lost. We could also reduce range length because I hope head will move forward continuously so big range is not needed.
I didn't think it through much so feel free to merge as is :)

Reply:
Alright, increased to 25.
Most witnesses will be smaller than 40MB, so it's unlikely that we'll ever hit full capacity anyway.